include/extractor/suffix_table.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef OSRM_EXTRACTOR_SUFFIX_LIST_HPP_
+#define OSRM_EXTRACTOR_SUFFIX_LIST_HPP_
+
+#include <string>
+#include <unordered_set>
+
+struct lua_State;
+
+namespace osrm
+{
+namespace extractor
+{
+// A table containing suffixes.
+// At the moment, it is only a front for an unordered set. At some point we might want to make it
+// country dependent and have it behave accordingly
+class SuffixTable final
+{
+  public:
+    // has to be called prior to suffix detection
+    void fill(lua_State *lua_state);
```

Comment:
Then why not passing the state to the ctor? => ctor establishes invariant, user is able to call member functions